### ISSUE TITLE
Use aria live regions so that inline error messages are announced by screen readers

### ISSUE THREAD

**Issue by chrismlusk** [verbb/formie#1650]
### What are you trying to do?

When field errors are inserted into the form (e.g., with Validate When Typing enabled), those messages are not announced by screen readers. 

### What's your proposed solution?

I am not an accessibility expert, but my understanding is that the error element generated with JavaScript should be an [ARIA Live Region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) with the `aria-live` attribute. For example:

```html
<div
    aria-live="assertive"
    data-error-message
    id="fui-error-fields[demo]"
>
```

However, this raises a couple of related concerns. 

First, should this be baked in by default? I think yes, but if so, should form editors have the ability to choose the politeness setting (e.g., `off`, `polite`, `assertive`)? That makes a potential update more complex, so maybe a default of `polite` for now would be fine?

Second, given this visual treatment ...

<img width="389" alt="error field" src="https://github.com/verbb/formie/assets/5922913/7f9de39f-862c-490b-97a3-46a91aa8d8a8">

... I believe the preferred error message experience for screen readers would be markup such as:

```html
<div aria-live="assertive" aria-atomic="true" id="fui-error-fields[username]">
    <svg aria-hidden="true"> ... </svg>
    <span class="sr-only">Error: </span>
    Enter your username.
</div>
```

That way, both sighted and screen-reader users have the same experience, with either the icon to communicate "error" to sighted users or the sr-only text being announced to screen-reader users.

However, I can't achieve this markup with AJAX-based forms. This relates to #1335.

Any thoughts how we could address this?

**Comment by engram-design** [comment]
Good call on this, forgot about `aria-live`. This will probably have to come with the validation revamp in https://github.com/verbb/formie/issues/556

I'll consider introducing icons for errors, and you're right that there are considerations for the politeness setting. I'm not _quite_ sure how I want to handle this, as that sort of setting is really only going to be touched (from the default) by clients who really know their accessibility stuff.

Excellent points though, thanks!

**Comment by chrismlusk** [comment]
It looks like #556 is a large task that's been around a while. Is that in progress or is it on the roadmap?

I have a client where we are addressing accessibility issues in response to a legal complaint, so getting error messages announced by screen readers is unfortunately a bit urgent. Would you consider adding `aria-live` to the [current error message node](https://github.com/verbb/formie/blob/craft-4/src/web/assets/frontend/src/js/utils/bouncer.js#L501-L505) in the meantime?

**Comment by engram-design** [comment]
No progress (yet) but being a breaking change, we want to roll it into a new major version. With Formie 3 slates for next year with Craft 5, we're going to do our _very_ best to fit it in there.

More than happy to add that as a stop-gap for now, but I'd like to revisit this some a more all-encompassing solution.

Are you happy to use `polite` or would you need this configurable?

**Comment by chrismlusk** [comment]
Understandable. Yes, `polite` is perfect and a reasonable default. Thank you for the quick help!

**Comment by engram-design** [comment]
Added for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.0.43"`.

**Comment by engram-design** [comment]
Added in [2.0.44](https://github.com/verbb/formie/releases/tag/2.0.44)
